packages/components/src/gradient-picker/index.js

```diff
@@ -105,10 +117,9 @@ export default function GradientPicker( {
 		() => onChange( undefined ),
 		[ onChange ]
 	);
-	const Component =
-		gradients?.length && gradients[ 0 ].gradients
-			? MultipleOrigin
-			: SingleOrigin;
+	const Component = isMultipleOriginArray( gradients )
+		? MultipleOrigin
+		: SingleOrigin;
```

Comment:
Applied the same logic used for `ColorPalette` also to `GradientPicker`, with the only difference that instead of checking for `colors` and `color` properties, we check for `gradients` and `gradient`